src/main/java/nextstep/sessions/infrastructure/JdbcSessionRepository.java

```diff
@@ -0,0 +1,114 @@
+                    new Duration(toLocalDateTime(rs.getTimestamp(5)), toLocalDateTime(rs.getTimestamp(6)))
+                )
+            )
+        );
+        return jdbcTemplate.queryForObject(sql, rowMapper, sessionId);
+    }
+
+    @Override
+    public CoverImage findCoverImageBySessionId(int sessionId) {
+        String sql = "select session_id, type, file_size, width, height from cover_image where session_id = ?";
+        RowMapper<CoverImage> rowMapper = (rs, rowNum) -> new CoverImage(
+            ImageType.valueOf(rs.getString(2)),
+            new ImageSize(rs.getInt(3), rs.getInt(4), rs.getInt(5))
+        );
+        return jdbcTemplate.queryForObject(sql, rowMapper, sessionId);
+    }
+
+    @Override
+    public List<Registration> findAllRegistrations(int sessionId) {
+        String sql = "select s.paid_type, s.fee, s.capacity, s.state, s.start_date, s.end_date, u.id, u.user_id, u.password, u.name, u.email from registration r join session s on r.session_id = s.id left join ns_user u on r.user_id = u.id where r.session_id = ?";
+        RowMapper<Registration> rowMapper = (rs, rowNum) -> new Registration(
+            new Session(
+                new SessionInfo(
+                    new EnrollmentInfo(
+                        new SessionType(PaidType.valueOf(rs.getString(1)), rs.getLong(2), rs.getInt(3)),
+                        SessionState.valueOf(rs.getString(4))
+                    ),
+                    new OpenInfo(
+                        new Duration(toLocalDateTime(rs.getTimestamp(5)), toLocalDateTime(rs.getTimestamp(6)))
+                    )
+                )
+            ),
+            new NsUser(rs.getLong(7), rs.getString(8), rs.getString(9), rs.getString(10), rs.getNString(11)),
+            new Payment()
+        );
+        return jdbcTemplate.query(sql, rowMapper, sessionId);
+    }
+
+
+    @Override
+    public void saveRegistration(int sessionId, Registration registration) {
+        String sql = "insert into registration (session_id, user_id, payment_id) values(?, ?, ?)";
+        jdbcTemplate.update(sql,
+            sessionId, registration.userPaymentInfo().user().getId(), registration.userPaymentInfo().payment().getId()
+        );
+    }
+
+    @Override
+    public int saveSession(Session session) {
+        String sql = "insert into session (paid_type, fee, capacity, state, start_date, end_date) values(?, ?, ?, ?, ?, ?)";
+        return jdbcTemplate.update(sql,
+            session.sessionInfo().enrollmentInfo().sessionType().payInfo().paidType().name(),
+            session.sessionInfo().enrollmentInfo().sessionType().payInfo().fee(),
+            session.sessionInfo().enrollmentInfo().sessionType().capacity(),
+            session.sessionInfo().enrollmentInfo().sessionState().name(),
+            Timestamp.valueOf(session.sessionInfo().openInfo().duration().startDate()),
+            Timestamp.valueOf(session.sessionInfo().openInfo().duration().endDate())
+        );
```

Comment:
> 인스턴스 변수를 2개로 유지하다보니 파라미터를 매핑할 때 getter들이 많이 나오고,

도메인 객체의 상태를 저장하거나 보여주려면 getter를 피할 수는 없다고 생각해요
디미터 법칙을 활용해서 값을 꺼내는 코드를 줄여볼 수 있을 것 같아요
DTO 등을 활용해볼 수도 있을 것 같네요
```suggestion
        return jdbcTemplate.update(sql,
            session.paidTypeName(),
            session.fee(),
            session.capacity(),
            ...
        );
```

Reply:
좋은 의견 감사합니다!
디미터 법칙을 적용하여 Session에 getter 메시지를 보내고, 나머지 연쇄적인 getter는 session에서만 처리하도록 했습니다.